framework/src/main/java/org/checkerframework/common/accumulation/AccumulationChecker.java

```diff
@@ -10,20 +13,71 @@
  * An accumulation checker is one that accumulates some property: method calls, map keys, etc.
  *
  * <p>This class provides a basic accumulation analysis that can be extended to implement an
- * accumulation type system. This accumulation analysis represents all facts as Strings. It
- * automatically includes returns-receiver aliasing to precisely handle fluent APIs, but otherwise
- * uses no alias analysis.
+ * accumulation type system. This accumulation analysis represents all facts as Strings.
+ *
+ * <p>This class supports modular alias analyses. To choose the alias analyses that your
+ * accumulation checker uses, override the {@link #createAliasAnalyses()} method. By default, the
+ * only alias analysis used is Returns Receiver.
  *
  * <p>The primary extension point is the constructor of {@link AccumulationAnnotatedTypeFactory},
  * which every subclass should override to provide custom annotations.
  */
 public abstract class AccumulationChecker extends BaseTypeChecker {
 
+    /**
+     * Set of alias analyses that are enabled in this particular accumulation checker. Set to
+     * non-null the first time {@link #isEnabled(AliasAnalysis)} is called, to allow the
+     * implementation of {@link #createAliasAnalyses()} to be as simple as possible.
+     */
+    private @MonotonicNonNull EnumSet<AliasAnalysis> aliasAnalyses;
+
     @Override
     protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
         LinkedHashSet<Class<? extends BaseTypeChecker>> checkers =
                 super.getImmediateSubcheckerClasses();
-        checkers.add(ReturnsReceiverChecker.class);
+        AliasAnalysis[] aliasAnalyses = createAliasAnalyses();
+        for (AliasAnalysis aliasAnalysis : aliasAnalyses) {
+            switch (aliasAnalysis) {
+                case RETURNS_RECEIVER:
+                    checkers.add(ReturnsReceiverChecker.class);
+                    break;
+            }
+        }
         return checkers;
     }
+
+    /**
+     * The alias analyses that an accumulation checker can support. To add support for a new alias
+     * analysis, add a new item to this enum, modify the loop in #getImmediateSubcheckerClasses, and
+     * then implement whatever functionality in the annotated type factory or transfer function that
+     * your new alias analysis permits.
+     */
+    public enum AliasAnalysis {
+        RETURNS_RECEIVER
+    }
+
+    /**
+     * Get the alias analyses that this checker should employ.
+     *
+     * @return the alias analyses
+     */
+    protected AliasAnalysis[] createAliasAnalyses(@UnderInitialization AccumulationChecker this) {
```

Comment:
Should this return an `EnumSet`?

Reply:
It returns an array rather than an enumset by design, so that overriding it is as easy as possible. With this signature, you can use the array constant syntax (as this version of the method does), which means that regardless of how many alias analyses you want to enable the method will always be very short and understandable. An enumset wouldn't be that much harder, of course, so if we think that's a lot clearer than I can change it.

Reply:
This is fine as is I think

Reply:
I had the same question as Manu when I read the code.  I think that using an array makes this less rather than more concise -- the `EnumSet.of()` and `EnumSet.allOf()` calls are slightly more concise.  Furthermore, it feels a bit convoluted, and clutters the code, to use an array that will be immediately converted to an EnumSet.

Reply:
Since this was confusing for both of you, I'll change it to use an EnumSet throughout.